File: pascal/operators/operator_modulo.pas
program OperatorModulo;

var
  i1, i2: integer;
  r1, r2: real;

begin
  i1 := 6; i2 := 1;
  writeln(i1, ' mod ', i2, ' = ', (i1 mod i2));
  i1 := 6; i2 := 2;
  writeln(i1, ' mod ', i2, ' = ', (i1 mod i2));
  i1 := 6; i2 := 3;
  writeln(i1, ' mod ', i2, ' = ', (i1 mod i2));
  i1 := 6; i2 := 4;
  writeln(i1, ' mod ', i2, ' = ', (i1 mod i2));
  i1 := 6; i2 := 5;
  writeln(i1, ' mod ', i2, ' = ', (i1 mod i2));
  i1 := 6; i2 := 6;
  writeln(i1, ' mod ', i2, ' = ', (i1 mod i2));

  r1 := 6; r2 := 3;
  // writeln(r1, ' mod ', r2, ' = ', (r1 mod r2)); // error
end.

File: java/operators/OperatorModulo.java
class OperatorModulo {
  public static void main(String[] args) {
    int i1, i2;

    i1 = 6; i2 = 1;
    System.out.println(String.format("%d %% %d = %d", i1, i2, (i1 % i2)));
    i1 = 6; i2 = 2;
    System.out.println(String.format("%d %% %d = %d", i1, i2, (i1 % i2)));
    i1 = 6; i2 = 3;
    System.out.println(String.format("%d %% %d = %d", i1, i2, (i1 % i2)));
    i1 = 6; i2 = 4;
    System.out.println(String.format("%d %% %d = %d", i1, i2, (i1 % i2)));
    i1 = 6; i2 = 5;
    System.out.println(String.format("%d %% %d = %d", i1, i2, (i1 % i2)));
    i1 = 6; i2 = 6;
    System.out.println(String.format("%d %% %d = %d", i1, i2, (i1 % i2)));

    float f1, f2;

    f1 = 6f; f2 = 3f;
    System.out.println(String.format("%f %% %f = %f", f1, f2, (f1 % f2)));
  }
}
